antlr/src/main/antlr/FSH.g4

```diff
@@ -10,7 +10,7 @@ alias:              KW_ALIAS SEQUENCE EQUAL (SEQUENCE | CODE);
 
 profile:            KW_PROFILE name sdMetadata+ sdRule*;
 extension:          KW_EXTENSION name (sdMetadata | context)* sdRule*;
-logical:            KW_LOGICAL name sdMetadata* lrRule*;
+logical:            KW_LOGICAL name (sdMetadata | characteristics)* lrRule*;
```

Comment:
I wondered if we should also support this on `resource` -- but after looking at some R5 resources, it looks like FHIR core does not use it on resources, so it probably isn't necessary. Maybe there are just assumptions that can be made about resources in general (like they always can be a target). Anyway... I think we're good!